middleware/proxy/reverseproxy.go

```diff
@@ -153,8 +155,37 @@ func (p *ReverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request, extr
 
 	copyHeader(rw.Header(), res.Header)
 
-	rw.WriteHeader(res.StatusCode)
-	p.copyResponse(rw, res.Body)
+	if res.StatusCode == HTTPSwitchingProtocols && outreq.Header.Get("Upgrade") == "websocket" {
```

Comment:
Doing a little more research... [Section 4.2 of the RFC](https://tools.ietf.org/html/rfc6455#section-4.2) says:

>    Servers MAY offload the management of the connection to other agents
   on the network, for example, load balancers and reverse proxies.  In
   such a situation, the server for the purposes of this specification
   is considered to include all parts of the server-side infrastructure
   from the first device to terminate the TCP connection all the way to
   the server that processes requests and sends responses.
> 
> EXAMPLE: A data center might have a server that responds to WebSocket
   requests with an appropriate handshake and then passes the connection
   to another server to actually process the data frames.  For the
   purposes of this specification, the "server" is the combination of
   both computers.

This makes me think that the reverse proxy should be invisible to the client, in other words, should return a 101 status and include that Upgrade header.

If you agree, I'm happy to make the change, or you can if you'd like. Just let me know what you think.